lib/koans/10_processes.ex

```diff
@@ -11,30 +11,87 @@ defmodule Processes do
     assert information[:status] == ___
   end
 
+  koan "Processes are referenced by their process ID (pid)" do
+    assert is_pid(self) == ___
+  end
+
   koan "New processes are spawned functions" do
     pid = spawn(fn -> nil end)
 
     assert Process.alive?(pid) == ___
   end
 
-  koan "You can send messages to processes" do
+  koan "Processes can send and receive messages" do
     send self, "hola!"
 
     receive do
       msg -> assert msg == ___
     end
   end
 
+  koan "Received messages are queued, first in first out" do
+    send self, "hola!"
+    send self, "como se llama?"
+
+    assert_receive ___
+    assert_receive ___
+  end
+
   koan "A common pattern is to include the sender in the message, so that it can reply" do
-    pid = spawn(fn -> receive do
-                         {:hello, sender} -> send sender, :how_are_you?
-                      end
-                 end)
+    greeter = fn ->
+      receive do
+        {:hello, sender} -> send sender, :how_are_you?
+      end
+    end
+
+    pid = spawn(greeter)
 
     send pid, {:hello, self}
     assert_receive ___
   end
 
+  def yelling_echo_loop do
+    receive do
+      {caller, value} ->
+        send caller, String.upcase(value)
+        yelling_echo_loop
+    end
+  end
+
+  koan "Use tail recursion to receive multiple messages" do
+    pid = spawn_link(&yelling_echo_loop/0)
```

Comment:
I went ahead and replace the `Process.exit` with `spawn_link`. Can revert if needed @felipesere @ukutaht